src/CloudFlare/IpRewrite.php

```diff
@@ -2,65 +2,78 @@
 
 namespace CloudFlare;
 
-class IpRewrite {
-    static protected $is_cf = NULL;
-    static protected $original_ip = FALSE;
-    static protected $rewritten_ip = FALSE;
-    
-    static protected $cf_ipv4 = array(
-        "199.27.128.0/21",
-        "173.245.48.0/20",
-        "103.21.244.0/22",
-        "103.22.200.0/22",
-        "103.31.4.0/22",
-        "141.101.64.0/18",
-        "108.162.192.0/18",
-        "190.93.240.0/20",
-        "188.114.96.0/20",
-        "197.234.240.0/22",
-        "198.41.128.0/17",
-        "162.158.0.0/15",
-        "104.16.0.0/12",
-        "172.64.0.0/13"
+require_once 'vendor/autoload.php';
+
+class IpRewrite
+{
+    protected static $is_loaded = false;
+    protected static $original_ip = null;
+    protected static $rewritten_ip = null;
+
+    // Found at https://www.cloudflare.com/ips/
+    protected static $cf_ipv4 = array(
+        '199.27.128.0/21',
+        '173.245.48.0/20',
+        '103.21.244.0/22',
+        '103.22.200.0/22',
+        '103.31.4.0/22',
+        '141.101.64.0/18',
+        '108.162.192.0/18',
+        '190.93.240.0/20',
+        '188.114.96.0/20',
+        '197.234.240.0/22',
+        '198.41.128.0/17',
+        '162.158.0.0/15',
+        '104.16.0.0/12',
+        '172.64.0.0/13',
+        '131.0.72.0/22',
     );
-    
-    static protected $cf_ipv6 = array(
-        "2400:cb00::/32",
-        "2606:4700::/32",
-        "2803:f800::/32",
-        "2405:b500::/32",
-        "2405:8100::/32"
+
+    protected static $cf_ipv6 = array(
+        '2400:cb00::/32',
+        '2606:4700::/32',
+        '2803:f800::/32',
+        '2405:b500::/32',
+        '2405:8100::/32',
     );
-    
-    // Returns boolean
-    public static function isCloudFlare()
+
+    // Helper method for testing, should not be used in production
+    public static function reset()
     {
-        self::rewrite();
-        return self::$is_cf;
+        self::$is_loaded = false;
+        self::$original_ip = null;
+        self::$rewritten_ip = null;
     }
-    
-    // Returns IP Address or false on error
-    public static function getRewrittenIP()
+
+    // Returns boolean
+    public static function isCloudFlare()
     {
-        self::rewrite();
-        return self::$rewritten_ip;
+        if (!isset($_SERVER['HTTP_CF_CONNECTING_IP'])) {
+            return false;
+        }
+
+        return true;
     }
-    
-    // Returns IP Address or false on error
+
+    // Returns IP Address or null on error
     public static function getOriginalIP()
     {
-        self::rewrite();
+        // If $original_ip is not set, return the REMOTE_ADDR
+        if (!isset(self::$original_ip)) {
+            return $_SERVER['REMOTE_ADDR'];
```

Comment:
I didn't want to cause anymore side effects. 

So I had two options if `original_ip` is not set.
- `return false;`
- `return $_SERVER['REMOTE_ADDR'] ;`

I chose the second option since `original_ip` can only be set if a rewrite action happens. If a rewrite action didn't happen `$_SERVER['REMOTE_ADDR']` **must** be the original ip.